lib/Config.php

```diff
@@ -90,34 +94,96 @@ public function getSettings($userId) {
 	}
 
 	/**
-	 * @return string
+	 * @return string[]
 	 */
-	public function getStunServer() {
-		$config = $this->config->getAppValue('spreed', 'stun_servers', json_encode(['stun.nextcloud.com:443']));
-		$servers = json_decode($config, true);
+	public function getAllServerUrlsForCSP(): array {
+		$urls = [];
+
+		foreach ($this->getStunServers() as $server) {
+			$urls[] = $server;
+		}
 
-		if ($servers === null) {
-			return $config ?: 'stun.nextcloud.com:443';
+		foreach ($this->getTurnServers() as $server) {
+			$urls[] = $server['server'];
 		}
 
-		if (is_array($servers) && !empty($servers)) {
+		foreach ($this->getSignalingServers() as $server) {
+			$urls[] = $this->getWebSocketDomainForSignalingServer($server['server']);
+		}
+
+		return $urls;
+	}
+
+	protected function getWebSocketDomainForSignalingServer(string $url): string {
```

Comment:
The `substr` expressions do not return the right domain if the signaling server does not include a trailing _/_.